### ISSUE TITLE
model_version="gpt-4"

### ISSUE THREAD

**Issue by rajayyala** [mljar/plotai#5]
Hi,

I am trying to set model version to gpt-4 as you suggested in the README file but I am getting the following error:

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[4], [line 6](vscode-notebook-cell:?execution_count=4&line=6)
      [3](vscode-notebook-cell:?execution_count=4&line=3) df = pd.DataFrame({"x":[1,2,3], "y": [4,5,6]})
      [4](vscode-notebook-cell:?execution_count=4&line=4) ## create a plot
----> [6](vscode-notebook-cell:?execution_count=4&line=6) plot = PlotAI(df,model_version="gpt-4")
      [8](vscode-notebook-cell:?execution_count=4&line=8) plot.make("scatter plot")

TypeError: PlotAI.__init__() got multiple values for argument 'model_version'

Initially when i tried with the default version , i got the following error:

...
    [178](https://vscode-remote+ssh-002dremote-002b10-002e253-002e0-002e135.vscode-resource.vscode-cdn.net/home/ubuntu/Dev/plai_test_1/~/miniconda3/envs/plai1/lib/python3.10/json/encoder.py:178)     """
--> [179](https://vscode-remote+ssh-002dremote-002b10-002e253-002e0-002e135.vscode-resource.vscode-cdn.net/home/ubuntu/Dev/plai_test_1/~/miniconda3/envs/plai1/lib/python3.10/json/encoder.py:179)     raise TypeError(f'Object of type {o.__class__.__name__} '
    [180](https://vscode-remote+ssh-002dremote-002b10-002e253-002e0-002e135.vscode-resource.vscode-cdn.net/home/ubuntu/Dev/plai_test_1/~/miniconda3/envs/plai1/lib/python3.10/json/encoder.py:180)                     f'is not JSON serializable')

TypeError: Object of type DataFrame is not JSON serializable

Any suggestions how to fix the above errors? Thanks

**Comment by pplonski** [comment]
Hi @rajayyala,

Are you using version `0.0.3`?

**Comment by rajayyala** [comment]
that is correct...version 0.0.3..but i was able to solve the issue by passing it as keyword parameter..then i got past the error

like so:

plot = PlotAI(df=df,model_version="gpt-4")

that worked...

however i am running into another issue...

frequently the generated python code throws error saying that df is not defined.

this is the code i am trying:

 df = pd.DataFrame({"x":[1,2,3], "y": [4,5,6]})
## create a plot

plot = PlotAI(df=df,model_version="gpt-4")
# plot = PlotAI(df)

plot.make("scatter plot")

this is the prompt it generated:

## create some sample data
# df = pd.DataFrame({"x":[1,2,3], "y":[4,5,6]})
df = pd.DataFrame({"x":[1,2,3], "y": [4,5,6]})
## create a plot

plot = PlotAI(df=df,model_version="gpt-4")
# plot = PlotAI(df)

plot.make("scatter plot")

csv_file_to_load = "2024-04-15T01-57_export.csv"
df_1 = pd.read_csv(csv_file_to_load)
```

Output only Python code.

and this is the response:


  ```python
# import required dependencies
import matplotlib.pyplot as plt

# Provide the plot
plt.scatter(df['x'], df['y'])
plt.xlabel('x')
plt.ylabel('y')
plt.title('Scatter plot of x and y')
plt.show()
```

as u can see its missing the dataframe variable "df"

Error output:
 name 'df' is not defined

attaching some screenshots of the same:

![image](https://github.com/mljar/plotai/assets/164526199/2ddd6f76-7fec-4e6c-8dd1-46cc2e85c24f)

![image](https://github.com/mljar/plotai/assets/164526199/67d61bdf-b97e-44de-9ada-9f7041c8d066)

**Comment by rajayyala** [comment]
any suggestions? thanks

**Comment by pplonski** [comment]
Try to move data frame creation into separate cell.

**Comment by rajayyala** [comment]
did that...got the same error...

it works some times and doesnt work some times though...so not sure if having it in the same cell is the issue....

also when we do in a python script then the concept of cell does not arise...right? any thoughts?

**Comment by pplonski** [comment]
There was bug in the code. I fixed it and released new version `0.0.4`, please let me know if it works for you.
